### ISSUE TITLE
buildkite-agent artifact download doesn't detect S3 authorisation failures.

### ISSUE THREAD

**Issue by orj** [buildkite/agent#197]
If buildkite-agent artifact download/upload fails due to an S3 authorisation failure it should report and error return code.  

We noticed recently that buildkite-agent was just saving the following XML for atifacts when we tried to download artifacts as part of our build scripts:

```
<Error>
<Code>AccessDenied</Code>
<Message>Access Denied</Message>
<RequestId>76C9A3D18A3031AF</RequestId>
<HostId>
FR33G95JgZpyzKv+vVrlLoJfw7xnPYeJr1x8nwdvY4PDrCuSncZ1sWRqnnzw9aHj/Lhf3lZUqC4=
</HostId>
</Error>
```

**Comment by keithpitt** [comment]
Close this as it's fixed in the latest release `v2.1` of the agent. Thanks for the report @orj!

**Comment by orj** [comment]
:+1:
